src/sentry/static/sentry/app/views/organizationIntegrations/integrationListDirectory.tsx

```diff
@@ -421,7 +465,8 @@ export class IntegrationListDirectory extends AsyncComponent<
 
   renderBody() {
     const {orgId} = this.props.params;
-    const {displayedList, selectedCategory, list} = this.state;
+    const {list, searchInput, selectedCategory} = this.state;
+    const displayedList = this.getDisplayedList();
```

Comment:
It seems that every interaction with search bar or category select causes displayedList to be recalculated twice [1]. Also it recalculates on every rerender, which is quite often bc of requests being made on that page on an interval [2].

[1]
<img width="420" alt="Screen Shot 2020-05-06 at 11 01 09 AM" src="https://user-images.githubusercontent.com/10491193/81211988-0bd4ac00-8f89-11ea-9828-f179acdfad40.png">

[2]
<img width="676" alt="Screen Shot 2020-05-06 at 11 09 46 AM" src="https://user-images.githubusercontent.com/10491193/81212731-26f3eb80-8f8a-11ea-820a-cff7ac7a06eb.png">

Reply:
@mgaeta I think it might be better to hold this up in state and calculate on `onLoadAllEndpointsSuccess()`, `handleSearchChange` and `onCategorySelect`.

Reply:
👍